chain-impl-mockchain/src/block/version.rs

```diff
@@ -68,10 +70,26 @@ impl BlockVersion {
     }
 }
 
-#[derive(Debug, Clone, Copy, FromPrimitive, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    Display,
+    EnumString,
+    FromPrimitive,
+    IntoStaticStr,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+)]
 pub enum ConsensusVersion {
+    #[strum(to_string = "none")]
```

Comment:
is there a way to make this value not possible in the string? None is only valid when the system starts up, but it is not a valid value in the initial message

Reply:
Is it valid in genesis.yaml?

Reply:
No it is not.

Reply:
I'm starting to suspect that the design of ConsensusVersion may be broken. Is `None` variant valid in any context? If system starts up and the inintial message does not set consensus version of block0, the node is in state not accepted by multiple mechanisms.

Maybe the None variant should be removed completely and block0-consensus in initial message should be mandatory (either bft of genesis)?

Reply:
Yes, it seems you are right. None should be removed here. The value is mandatory.